xblockutils/public/studio_edit.js

```diff
@@ -65,6 +66,11 @@ function StudioEditableXBlockMixin(runtime, element) {
                 }
             });
         }
+
+        if (type == 'datepicker' && datepickerAvailable) {
+            $field.datepicker('destroy');
+            $field.datepicker({dateFormat: "m/d/yy"});
```

Comment:
I think you need to be consistent with the rest of the platform and leave it as is. Why don't you open a JIRA ticket to ask them to change to ISO 8601 in the future.